### PR TITLE
feat(argo-cd): make ServiceMonitor deployment conditional on CRD existence

### PR DESCRIPTION
**Pull request by alex-souslik-hs**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by yu-croco** [comment]
Hi  @alex-souslik-hs , thank you for your PR.
Can you please separate this PR into resources(e.g. argo-cd, argo-workflows)?
We make PR by each resource.
Ref: https://github.com/argoproj/argo-helm/pulls?q=is%3Apr+is%3Amereged

Also can you please give us the explanation of motivation for this change?
I am not familiar with it but does `monitoring.coreos.com/v1` have compatibility to specific k8s version?

**Comment by alex-souslik-hs** [comment]
> Hi @alex-souslik-hs , thank you for your PR. Can you please separate this PR into resources(e.g. argo-cd, argo-workflows)? We make PR by each resource, instead of bundling them. Ref: https://github.com/argoproj/argo-helm/pulls?q=is%3Apr+is%3Amereged

It's off course is possible but if you can make an exception for this particular case that'd be great. If not I'll work on splitting.
> Also can you please give us the explanation of motivation for this change? I am not familiar with it but does `monitoring.coreos.com/v1` have compatibility to specific k8s version (I didn't get info for this, so I want the detail) ?

`monitoring.coreos.com/v1` is not a part of the k8s API but comes from the Prometheus Operator. So for example if I deploy a new cluster and I want to install ArgoCD first and Prometheus after I need two value files for ArgoCD.

**Comment by yu-croco** [comment]
@alex-souslik-hs 

>@yu-croco Off course I can but if you can make an exception for this particular case that'd be great. If not I'll work on splitting.

Please separate PR. 

>monitoring.coreos.com/v1 is not a part of the k8s API but comes from the Prometheus Operator. So for example if I deploy a new cluster and I want to install ArgoCD first and Prometheus after I need two value files for ArgoCD.

Thank you for explaining!

**Comment by alex-souslik-hs** [comment]
@yu-croco cleaned up PR to be ArgoCD only, this is the one that was important to me anyway.

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: v2.7.7
 kubeVersion: ">=1.23.0-0"
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 5.39.0
+version: 5.39.1
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:6]: Please bump the minor version, as this is not a real bugfix
```suggestion
version: 5.40.0
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Almost LGTM

**Comment by alex-souslik-hs** [comment]
@mkilchhofer done.

**Review by yu-croco — APPROVED** [review]
LGTM

**Review by pdrastil — APPROVED** [review]
LGTM

**Review by mkilchhofer — APPROVED** [review]

**Comment by batazor** [comment]
@alex-souslik-hs Will this work correctly in case prometheus-operator is installed via argocd?  

Now I am installing in this sequence prometheus-operator-crd -> argocd -> prometheus-operator

Won't we have cases that argocd will be installed without installing monitoring components?

**Comment by alex-souslik-hs** [comment]
> @alex-souslik-hs Will this work correctly in case prometheus-operator is installed via argocd?
> 
> Now I am installing in this sequence prometheus-operator-crd -> argocd -> prometheus-operator
> 
> Won't we have cases that argocd will be installed without installing monitoring components?

@batazor 
If you pre-install the CRDs you'll have the same behavior as before, the idea is to allow you to install ArgoCD first and deploy everything else with it without modifying its values.

@yuha0
It's important to note the optional dependency `redis-ha` chart works the same way, so it's possible you already had this behavior on some of your `ServiceMonitors`.

**Comment by FalconerTC** [comment]
I believe this change has caused my ArgoCD service monitors to disappear. I manage ArgoCD with ArgoCD and after this upgrade I am no loner seeing any ArgoCD Service Monitors (I am however seeing ServiceMonitor for argo rollouts, but I see that's because this capabilities check isn't set there). Is there something in particular I should do to mock `Capabilities.APIVersions.Has "monitoring.coreos.com/v1`  when managing argocd with argocd?

**Comment by mkilchhofer** [comment]
Yup some tools like Kustomize and Helmfile do not pass the available APIversions to helm. In your mentioned issue above, there is also a comment which explains how to pass APIversions:

> Example:
> 
> ```yaml
> namespace: foo
> 
> helmCharts:
> - name: bar
>   releaseName: bar
> 
>   #...
>   apiVersions:
>     - apiregistration.k8s.io/v1
>   #...
> ```

**Pull request merged**